syntax = "proto3";


option objc_class_prefix = "HLW";

package vehicle;

service DataExchanger {

  rpc ManageSpeed (SpeedAction) returns ( NumberDataReply) {}
  rpc GetNumberData (Ping) returns ( NumberDataReply) {}
  rpc FillFuel (FuelAmount) returns ( FuelAmount) {}
  rpc GetTotalFuel (Ping) returns ( FuelAmount) {}

}


message Ping {
    bool binary=1;
}

message FuelAmount {

     double fuel = 1;
}
enum SpeedType {
  LEVER_UP = 0;
  PRESERVE = 1;
  SLOW = 2;
  STOP = 3;
}

message SpeedAction {

     SpeedType speedType = 1;
}


message NumberDataReply {
    double rpm = 1;
    double speed = 2;
    double fuel = 3;
    double temperature = 4;
    double distance = 5;
}


